Public/Src/Engine/Scheduler/ProcessResourceManager.cs

```diff
@@ -270,7 +286,12 @@ public void RegisterQueryRamUsageMb(Func<int> queryRamUsageMb)
 
             public int QueryRamUsageMb()
             {
-                return m_queryRamUsageMb?.Invoke() ?? 0;

```

Comment:
>m_queryRamUsageMb [](start = 23, length = 17)

I think there needs to be lock here so we prevent calling this concurrently with dispose. Maybe capture the variable, check if not null, and lock on captured variable. Also do the same in dispose.